modules/apps/account/account-service/src/main/java/com/liferay/account/internal/model/listener/OrganizationModelListener.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.account.internal.model.listener;
+
+import com.liferay.account.model.AccountEntryOrganizationRel;
+import com.liferay.account.service.AccountEntryOrganizationRelLocalService;
+import com.liferay.portal.kernel.exception.ModelListenerException;
+import com.liferay.portal.kernel.model.BaseModelListener;
+import com.liferay.portal.kernel.model.ModelListener;
+import com.liferay.portal.kernel.model.Organization;
+
+import java.util.List;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Pei-Jung Lan
+ */
+@Component(immediate = true, service = ModelListener.class)
+public class OrganizationModelListener extends BaseModelListener<Organization> {
+
+	@Override
+	public void onAfterRemove(Organization organization)
+		throws ModelListenerException {
+
+		super.onAfterRemove(organization);
+
+		List<AccountEntryOrganizationRel> accountEntryOrganizationRels =
+			_accountEntryOrganizationRelLocalService.
+				getAccountEntryOrganizationRelsByOrganizationId(
+					organization.getOrganizationId());
+
+		for (AccountEntryOrganizationRel accountEntryOrganizationRel :
+				accountEntryOrganizationRels) {
+
+			_accountEntryOrganizationRelLocalService.
+				deleteAccountEntryOrganizationRel(accountEntryOrganizationRel);
+		}
```

Comment:
I wonder if we should just create a `delete*ByOrganizationId` so that all these delete calls fall under the same transaction boundary. Though if they already are inside the same transaction boundary because of being inside a ModelListener, then it's a moot point and we don't have to change anything. I'm not sure how to test this, but I thought I'd mention it for consideration.

Reply:
@Preston-Crary would you be willing to weigh in on this? I'm not sure whether or not these delete calls are under the same transaction or not. My _guess_ is that since they are calling through the `AccountEntryOrganizationRelLocalService`, they are each opening a new transaction.

Reply:
@drewbrokke you can check by looking at the Transactional annotation on the AccountEntryOrganizationRelLocalService.deleteAccountEntryOrganizationRel() method. It only creates a new one if it's `propagation = Propagation.REQUIRES_NEW`.  Since there's no annotation there it's using the class annotation.  You always have a parent transaction inside a model listener.

Reply:
Thanks @preston-crary.

If I understand it correctly, since `Propagation.REQUIRES_NEW` is not explicitly set, it uses the default `REQUIRED` propagation so all these delete calls should run in the same transaction in the model listener.

Reply:
@pei-jung that's correct.